### PR TITLE
Make buildifier sanity test fail-able

### PR DESCRIPTION
**Pull request by lidizheng**
I introduced sanity test for Bazel files months ago, but it is actually not a fail-able test. This PR fixes the script, and format all the Bazel files.

### CODE REVIEW

File: tools/run_tests/sanity/check_buildifier.sh [hunk 1]
```diff
@@ -13,13 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# The script to sanitize Bazel files
+# The script to check if Bazel files needs to be formatted.
```
Comment (jtattermusch) [inline tools/run_tests/sanity/check_buildifier.sh]: nit: needs -> need

File: tools/distrib/buildifier_format_code.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#! /bin/bash -ex
+#! /bin/bash -e
```
Comment (jtattermusch) [inline tools/distrib/buildifier_format_code.sh]: I think using `set -e` at the beginning of the script is much more readable. (No one looks above the license header).
Also below.

**Comment by lidizheng** [comment]
@gnossen I noticed this when I'm trying to write my all-in-one lint snippet.

**Comment by lidizheng** [comment]
Known failure: https://github.com/grpc/grpc/issues/20624

**Pull request merged**